ui/src/app/utils/retry.tsx

```diff
@@ -17,3 +20,70 @@ export async function fetchAbortableRetry<T>(fetchFn: () => Promise<T>, timeoutM
     }
   }
 }
+
+
+
+export async function fetchWithGlobalErrorHandler<T>(fetchFn: () => Promise<T>, maxRetries: number = 3): Promise<T> {
+  let retries = 0;
+  while (true) {
+    try {
+      const returnVal = await fetchFn();
+      console.log(returnVal);
```

Comment:
So this is going to log the returnVal if the api is successful , i wonder if we do want to print the success result of each API thats going to use this